inference-engine/src/multi_device/multi_device_exec_network.cpp

```diff
@@ -60,71 +90,241 @@ MultiDeviceExecutableNetwork::MultiDeviceExecutableNetwork(const DeviceMap<Infer
-        auto& idleWorkerRequests = _idleWorkerRequests[device];
-        workerRequests.resize(numRequests);
-        _inferPipelineTasksDeviceSpecific[device] = std::unique_ptr<ThreadSafeQueue<Task>>(new ThreadSafeQueue<Task>);
-        auto* idleWorkerRequestsPtr = &(idleWorkerRequests);
-        idleWorkerRequests.set_capacity(numRequests);
-        for (auto&& workerRequest : workerRequests) {
-            workerRequest._inferRequest = { network, network->CreateInferRequest() };
-            auto* workerRequestPtr = &workerRequest;
-            IE_ASSERT(idleWorkerRequests.try_push(workerRequestPtr) == true);
-            workerRequest._inferRequest->SetCallback(
-                [workerRequestPtr, this, device, idleWorkerRequestsPtr] (std::exception_ptr exceptionPtr) mutable {
-                    IdleGuard idleGuard{workerRequestPtr, *idleWorkerRequestsPtr};
-                    workerRequestPtr->_exceptionPtr = exceptionPtr;
-                    {
-                        auto capturedTask = std::move(workerRequestPtr->_task);
-                        capturedTask();
-                    }
-                    // try to return the request to the idle list (fails if the overall object destruction has began)
-                    if (idleGuard.Release()->try_push(workerRequestPtr)) {
-                        // let's try to pop a task, as we know there is at least one idle request, schedule if succeeded
-                        // if no device-agnostic tasks, let's try pop the device specific task, schedule if succeeded
-                        Task t;
-                        if (_inferPipelineTasks.try_pop(t))
-                            ScheduleToWorkerInferRequest(std::move(t));
-                        else if (_inferPipelineTasksDeviceSpecific[device]->try_pop(t))
-                            ScheduleToWorkerInferRequest(std::move(t), device);
-                    }
-                });
+            _networkFirstReady = _cpuFuture.get();
+            SetPerfCounts(_networkFirstReady);
+        } catch (const std::exception& e) {
+            printf("Warning: load network to CPU failed: %s\n", e.what());
+            _networkActualNeeded = _acceleratorFuture.get();
         }
+    } else if (_acceleratorFuture.valid()) {  // only accelerator is valid, like AUTO:GPU
+        _networkActualNeeded = _acceleratorFuture.get();
+    } else if (_cpuFuture.valid()) {  // only CPU is valid, like AUTO:CPU
+        _networkActualNeeded = _cpuFuture.get();
+    } else {
+        IE_THROW() << "No device task available";
+    }
+
+    // if there is only one device or loading CPU device is failed,
+    // the ActualNetwork is already ok now.
+    if (!_acceleratorFuture.valid()) {
+        _alreadyActualNetwork = true;
+        // setPerfcounts
+        SetPerfCounts(_networkActualNeeded);
+    }
+}
+
+bool MultiDeviceExecutableNetwork::IsActualNetworkReady() const {
+    if (_alreadyActualNetwork) {
+        WaitActualNetworkReady();
```

Comment:
moving the question from the original PR to here: why we need this call in this func?

Reply:
need to set needPerfCounters after loading Actual Network,  but I can't do it in async threads.

Reply:
I also had a question on the needPerfCounters in the original PR:
https://github.com/openvinotoolkit/openvino/pull/7076#discussion_r702985084
again, why we cannot pass the needPerfCounters to the MultiDeviceExecNetwork constructor? Why we need to check that here- this config option cannot change (if it is set by user, it is SAME for any device in AUTO/MULTI)